featuretools/entityset/entityset.py

```diff
@@ -469,31 +439,52 @@ def find_path(self, start_entity_id, goal_entity_id,
             else:
                 return []
 
-        # BFS so we get shortest path
-        start_node = BFSNode(start_entity_id, None, None)
-        queue = [start_node]
-        nodes = {}
+        # Search for bath using BFS to get the shortest path.
+        # Start by initializing the queue with all relationships from start entity
+        queue = [[r] for r in self.get_forward_relationships(start_entity_id)] + \
+                [[r] for r in self.get_backward_relationships(start_entity_id)]
```

Comment:
Would using something like `collections.deque` instead of a list here be worth doing?

Reply:
i think the main reason to use deque would be for performance, but i dont think performance is a problem since this iterating over relationships in dataset, which can't be that large of a list.